manageiq-operator/pkg/controller/manageiq/manageiq_controller.go

```diff
@@ -137,47 +135,92 @@ func (r *ReconcileManageIQ) Reconcile(request reconcile.Request) (reconcile.Resu
 
 func (r *ReconcileManageIQ) generateHttpdResources(cr *miqv1alpha1.ManageIQ) error {
 	httpdConfigMap := miqtool.NewHttpdConfigMap(cr)
-	if err := r.createk8sResIfNotExist(cr, httpdConfigMap, &corev1.ConfigMap{}); err != nil {
+
+	if _, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, httpdConfigMap, func() error {
```

Comment:
I was thinking the callback function could be returned by the `miqtool` functions, but if you would rather keep it here, can you move it out of the `CreateOrUpdate` call? I think it would make all of this easier to read.

Reply:
let me just put them in `miqtool` functions then